### PR TITLE
Fix GN dependencies in //brave/browser/download

### PR DESCRIPTION
**Pull request by mariospr**
This patch fixes this dependency without creating a dependency loop
by directly importing the sources and deps from browser/sources.gni,
which in turn gets included by //chrome/browser where needed.

On top of that, this CL also cleans up some includes that were not
pointing to the required headers, but to other headers that would
indirectly pull the one actually required.

Resolves https://github.com/brave/brave-browser/issues/11881

## Submitter Checklist:

- [x] I confirm that no security/privacy review [is needed](https://github.com/brave/handbook/blob/master/development/security.md#when-is-a-security-review-needed), or that I have [requested](https://github.com/brave/security/issues/new/choose) one
- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [x] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [x] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [x] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [x] Checked the PR locally: `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests`, `npm run lint`, `npm run gn_check`, `npm run tslint`
- [x] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/handbook/blob/master/development/security.md#when-is-a-security-review-needed), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

### CODE REVIEW

File: browser/download/sources.gni
Comment (bridiver) [inline browser/download/sources.gni:10]: this shouldn't be relevant anymore, we should only include the direct deps for headers in brave_download_item_model.h/.cc

Comment (mariospr) [reply browser/download/sources.gni]: Yes, you're right and I should have noticed before proposing this PR.

Changed it now to remove that dependency on //chrome/common/safe_browsing:proto as well as the one on //chrome/common, and added a new one for //services/network/public/cpp now.

@bridiver PTAL, thanks

**Comment by mariospr** [comment]
@bridiver Ping on this for a second review, thanks

**Comment by mariospr** [comment]
@bridiver Pinging again now that I've rebased this on top of the big PR that just landed on master.

@iefremov Adding you too as well in case you want to take a look and/or clear this PR from Brian's review queue

File: browser/download/brave_download_item_model.h
Comment (iefremov) [inline browser/download/brave_download_item_model.h:9]: we could replace it with a forward declaration

**Review by iefremov — APPROVED** [review]

File: browser/download/brave_download_item_model.h
Comment (mariospr) [reply browser/download/brave_download_item_model.h]: Oops! Fixed now, thanks!

Additionally, making the most that I'm adding two items to the two lists at the bottom of //brave/browser/sources.gni, I've added one minor commit on top of this PR that normalizes & sorts those lists, which will make things easier for future additions to the file.

Will merge once CI confirms it's happy.

**Comment by mariospr** [comment]
Mac failure is unrelated, so merging...

**Pull request merged**
